src/System Application/App/File System/src/Account/FileAccountImpl.Codeunit.al

```diff
@@ -0,0 +1,247 @@
+        NewDefaultFileAccount: Record "File Account";
+        FileScenario: Codeunit "File Scenario";
+    begin
+        GetAllAccounts(false, AllFileAccounts);
+
+        if AllFileAccounts.IsEmpty() then
+            exit; //All of the accounts were deleted, nothing to do
+
+        if AllFileAccounts.Get(CurrentDefaultAccountId, Connector) then
+            exit; // The default account was not deleted or it never existed
+
+        // In case there's only one account, set it as default
+        if AllFileAccounts.Count() = 1 then begin
+            MakeDefault(AllFileAccounts);
+            exit;
+        end;
+
+        Commit();  // Commit the accounts deletion in order to prompt for new default account
+        if PromptNewDefaultAccountChoice(NewDefaultFileAccount) then
+            MakeDefault(NewDefaultFileAccount)
+        else
+            FileScenario.UnassignScenario(Enum::"File Scenario"::Default); // remove the default scenario as it is pointing to a non-existent account
+    end;
+
+    local procedure PromptNewDefaultAccountChoice(var NewDefaultFileAccount: Record "File Account"): Boolean
+    var
+        FileAccountsPage: Page "File Accounts";
+    begin
+        FileAccountsPage.LookupMode(true);
+        FileAccountsPage.EnableLookupMode();
+        FileAccountsPage.Caption(ChooseNewDefaultTxt);
+        if FileAccountsPage.RunModal() = Action::LookupOK then begin
+            FileAccountsPage.GetAccount(NewDefaultFileAccount);
+            exit(true);
+        end;
+
+        exit(false);
+    end;
+
+    local procedure ImportLogo(var FileAccount: Record "File Account"; Connector: Interface "File System Connector")
+    var
+        FileConnectorLogo: Record "File System Connector Logo";
+        TempBlob: Codeunit "Temp Blob";
+        Base64Convert: Codeunit "Base64 Convert";
+        ConnectorLogoBase64: Text;
+        OutStream: Outstream;
+        InStream: InStream;
+        ConnectorLogoDescriptionTxt: Label '%1 Logo', Locked = true;
+    begin
+        ConnectorLogoBase64 := Connector.GetLogoAsBase64();
+
+        if ConnectorLogoBase64 = '' then
+            exit;
+        if not FileConnectorLogo.Get(FileAccount.Connector) then begin
+            TempBlob.CreateOutStream(OutStream);
+            Base64Convert.FromBase64(ConnectorLogoBase64, OutStream);
+            TempBlob.CreateInStream(InStream);
+            FileConnectorLogo.Connector := FileAccount.Connector;
+            FileConnectorLogo.Logo.ImportStream(InStream, StrSubstNo(ConnectorLogoDescriptionTxt, FileAccount.Connector));
+            if FileConnectorLogo.Insert() then;
+        end;
+        FileAccount.Logo := FileConnectorLogo.Logo
+    end;
+
+    procedure IsAnyAccountRegistered(): Boolean
+    var
+        FileAccount: Record "File Account";
+    begin
+        GetAllAccounts(false, FileAccount);
+
+        exit(not FileAccount.IsEmpty());
+    end;
+
+    internal procedure IsUserFileAdmin(): Boolean
```

Comment:
what's special about this method that it is marked internal in an internal codeunit?

Reply:
Other question, why should it be part of the interface? It is internal and not local because the Wizard Page needs it.

Reply:
the codeunit is marked as internal so everything in it is internal. Setting the procedure itself to internal does nothing.